src/main/java/org/wso2/carbon/connector/amazons3/auth/AmazonS3URLSignatureCreation.java

```diff
@@ -104,8 +103,8 @@ public final void connect(final MessageContext messageContext) {
         } catch (ParseException pe) {
             storeErrorResponseStatus(messageContext, pe, AmazonS3Constants.INVALID_KEY_ERROR_CODE);
             handleException("Could not parse the date in correct format", pe, messageContext);
-        } catch (RegistryException re) {
-            handleException("No File in Registry", re, messageContext);
+        } catch (Exception e) {
```

Comment:
Using a generic exception because we need to make it compatible with both EI and MI, where MI runtime does not contain   `org.wso2.carbon.registry.core.exceptions.RegistryException`